test/functionalTest/cases/0787_cprs_full_functional/update_shadowed_attribute_in_ngsiv2.test

```diff
@@ -0,0 +1,320 @@
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Update With Shadowed Attribute
+
+--SHELL-INIT--
+dbInit CB
+dbInit CP1
+brokerStart CB
+brokerStart CP1
+
+--SHELL--
+
+#
+# 01. Update/APPEND E1/T1/A1 in CP1
+# 02. Update/APPEND E1/T1/A1 in CB
+# 03. Register E1/T1/A1 in CB, provApp: CP1
+# 04. Update/UPDATE E1/T1/A1 in CB
+# 05. Query E1/T1/A1 in CP1 (untouched by 04)
+# 06. Query E1/T1/A1 in CB (modified by 04)
+#
+
+echo "01. Update/APPEND E1/T1/A1 in CP1"
+echo "================================="
+payload='{
+  "contextElements": [
+    {
+      "type": "T1",
+      "id":   "E1",
+      "attributes": [
+        {
+          "name": "A1",
+          "type": "string",
+          "value": "A1 in CP1"
+        }
+      ]
+    }
+  ],
+  "updateAction": "APPEND"
+}'
+orionCurl --url /v1/updateContext --payload "$payload" --port $CP1_PORT
+echo
+echo
+
+
+echo "02. Update/APPEND E1/T1/A1 in CB"
+echo "================================"
+payload='{
+  "contextElements": [
+    {
+      "type": "T1",
+      "id":   "E1",
+      "attributes": [
+        {
+          "name": "A1",
+          "type": "string",
+          "value": "A1 in CB"
+        }
+      ]
+    }
+  ],
+  "updateAction": "APPEND"
+}'
+orionCurl --url /v1/updateContext --payload "$payload"
+echo
+echo
+
+
+echo "03. Register E1/T1/A1 in CB, provApp: CP1"
+echo "========================================="
+payload='{
+  "contextRegistrations": [
+  {
+    "entities": [
+      {
+         "type": "T1",
+         "isPattern": "false",
+         "id": "E1"
+      }
+    ],
+    "attributes": [
+      {
+        "name": "A1",
+        "type": "string",
+        "isDomain": "false"
+      }
+    ],
+    "providingApplication": "http://localhost:'${CP1_PORT}'/v1"
+    }
+ ],
+ "duration": "P1M"
+}'
+orionCurl --url /v1/registry/registerContext --payload "$payload"
+echo
+echo
+
+
+echo "04. Update/UPDATE E1/T1/A1 in CB"
+echo "================================"
+payload='{
+  "A1": "A1 modified by 04"
+}'
+orionCurl --url '/v2/entities/E1/attrs?type=T1&options=keyValues' --payload "$payload"
```

Comment:
I'd suggest to use PATCH in this case. That case, you fully ensure that you are doing an update (POST could cause append). The result of the test should be the same with PATCH.